spec/lib/miq_automation_engine/miq_ae_method_dispatch_spec.rb

```diff
@@ -0,0 +1,105 @@
+require "spec_helper"
+include AutomationSpecHelper
+
+describe "MiqAeMethodDispatch" do
+  before do
+    @method_name     = 'MY_METHOD'
+    @method_instance = 'MY_METHOD_INSTANCE'
+    @method_class    = 'MY_METHOD_CLASS'
+    @domain          = 'SPEC_DOMAIN'
+    @namespace       = 'NS1'
+    @root_class      = "TOP_OF_THE_WORLD"
+    @root_instance   = "EVEREST"
+    @automate_args   = {:namespace        => @namespace,
+                        :class_name       => @root_class,
+                        :instance_name    => @root_instance,
+                        :automate_message => 'create'}
+    MiqServer.stub(:my_zone).and_return('default')
+    @pidfile = File.join(Dir.mktmpdir, "rip_van_winkle.pid")
+    clear_domain
+  end
+
+  def clear_domain
+    MiqAeDomain.find_by_name(@domain).try(:destroy)
+  end
+
+  def rip_van_winkle_script
+    <<-'RUBY'
+      pidfile    = $evm.inputs['pidfile']
+      File.open(pidfile, 'w') { |file| file.write(Process.pid.to_s) }
+      10.times do
+          STDERR.puts "Hello from stderr channel"
+          STDOUT.puts "Hello from stdout channel"
+      end
+      sleep(600)
```

Comment:
@Fryguy 
This method should technically end after 10 minutes but it gets terminated by the server since it doesn't end when the timer pops from the queue which kills the automate request and any threads and automate methods that are hung.
The whole spec ends in 4 seconds since the queue msg_timeout is set to 2 seconds.